attestation-service/src/verifier/cca/mod.rs

```diff
@@ -77,16 +85,42 @@ impl Verifier for CCA {
             .with_new_session_url(api_endpoint)
             .build()?;
 
-        let nonce = Nonce::Size(32);
+        let mut hasher = Sha384::new();
+        hasher.update(&nonce);
+        hasher.update(&attestation.tee_pubkey.k_mod);
+        hasher.update(&attestation.tee_pubkey.k_exp);
+        let mut hash_of_nonce_pubkey = hasher.finalize().to_vec();
+        hash_of_nonce_pubkey.resize(64, 0);
+
+        log::info!(
+            "HASH(nonce||pubkey):\n\t{}\n",
+            hex::encode(&hash_of_nonce_pubkey)
+        );
+
         let token = evidence.token;
-        match cr.run(nonce, my_evidence_builder, token.clone()).await {
+        let n = Nonce::Value(hash_of_nonce_pubkey.clone());
+        let result = match cr.run(n, my_evidence_builder, token.clone()).await {
             Err(e) => {
                 log::error!("Error: {}", e);
-                return Err(anyhow!("Attestation failed with error: {:?}", e));
-            }
-            Ok(attestation_result) => {
-                log::info!("Attestation Result: {}", attestation_result)
+                panic!("Attestation failed with error: {:?}", e);
```

Comment:
Let's try not to use `panic` or `unwrap` inside the codes.
```suggestion
                bail!("CCA Attestation failed with error: {:?}", e);
```

Reply:
Done